File: src/FoodOrderSystem.App/ClientApp/src/app/dish-category/dish.model.ts
import {DishVariantModel} from './dish-variant.model';

export class DishModel {
  constructor(init?: Partial<DishModel>) {
    Object.assign(this, init);
    this.id = undefined;
  }

  public id: string;

  public name: string;

  public description: string;

  public productInfo: string;

  public variants: Array<DishVariantModel>;
}

File: src/FoodOrderSystem.App/Models/DishModel.cs
using System;
using System.Collections.Generic;

namespace FoodOrderSystem.App.Models
{
    public class DishModel
    {
        public Guid Id { get; set; }
        
        public string Name { get; set; }
        
        public string Description { get; set; }
        
        public string ProductInfo { get; set; }
        
        public int OrderNo { get; set; }
        
        public List<DishVariantModel> Variants { get; set; }
    }
}